packages/cubejs-schema-compiler/adapter/PreAggregations.js

```diff
@@ -201,6 +201,25 @@ class PreAggregations {
         timeDimensions.map(d => [d.dimension, d.granularity || 'date'])
       ) || [];
     }
+    // [TimeDimension] -> [TimeDimension]
+    function spawnTimeDimensions(timeDimensions = []) {
```

Comment:
This is cool approach! It would be even better if we can pre-generate possible time dimensions based on `transformedQuery` rather than `references`. This way it can be saved once for query and used for all pre-aggregations which is very important from performance perspective.

Reply:
Sharp eye! I will update accordingly.

Reply:
@paveltiunov 
Before actual implementation, I considered it again and found it not an easy change.
If every `timeDimension` in query can also be found in a pre-aggregation, we can say they are matched!
So expanding the pre-aggregation time dimension to match more queries is a natural approach.
If we keep `references` unchanged and turn around to update `transformedQuery `, the logic will become obscure and hard to read.
https://github.com/cube-js/cube.js/blob/85208b755c1385f26c9855dd3c8f6e10b02ed588/packages/cubejs-schema-compiler/adapter/PreAggregations.js#L242

Reply:
@Justin-ZS Yeah. Agree. But in case of this particular method we need to prioritize performance over readability unfortunately.